src/main/java/net/sf/jabref/model/metadata/MetaData.java

```diff
@@ -139,32 +70,20 @@ public void putData(String key, List<String> orderedData) {
      * returned by getGroups() so far!!!
      */
     public void setGroups(GroupTreeNode root) {
-        groupsRoot = root;
+        groupsRoot = Objects.requireNonNull(root);
+        groupsRoot.subscribeToDescendantChanged(groupTreeNode -> eventBus.post(new GroupUpdatedEvent(this)));
         eventBus.post(new GroupUpdatedEvent(this));
     }
 
     /**
      * @return the stored label patterns
      */
-    public AbstractBibtexKeyPattern getBibtexKeyPattern(GlobalBibtexKeyPattern globalPattern) {
-        if (bibtexKeyPattern != null) {
-            return bibtexKeyPattern;
-        }
+    public AbstractBibtexKeyPattern getCiteKeyPattern(GlobalBibtexKeyPattern globalPattern) {
```

Comment:
Since you are doing this for the other methods: Do a null check of `globalPattern`.